### ISSUE TITLE
WhereBetween ignore DB::raw expressions

### ISSUE THREAD

**Issue by rorc** [laravel/framework#20803]
- Laravel Version: 5.4.33

### Description:
Results of
```php
Order::whereBetween('created_at', [
    '2017-08-22 21:30:00',
    \DB::raw('date_add(created_at, interval 1 week)')
])->get();
```
Is
```mysql
select * from `orders`
where `created_at` between '2017-08-22 21:30:00'
    and 'date_add(created_at, interval 1 week)'
```

**Comment by Dylan-DPC** [comment]
yes this is intended behaviour. It is same with other functions such as `where`.

**Comment by staudenmeir** [comment]
This has been fixed in 5.6.24 (#24435).
